mipengine/controller/node_landscape_aggregator.py

```diff
@@ -274,67 +276,78 @@ def get_node_specific_datasets(
         )
 
 
-def _get_datasets_per_node(
-    nodes: Dict[str, NodeInfo],
-) -> Dict[str, Dict[str, Dict[str, str]]]:
-    datasets_per_node = {}
-    for node_id, node_info in nodes.items():
-        node_socket_addr = _get_node_socket_addr(node_info)
-        datasets_per_data_model = _get_node_datasets_per_data_model(node_socket_addr)
-        if datasets_per_data_model:
-            datasets_per_node[node_info.id] = datasets_per_data_model
-    return datasets_per_node
-
+def data_fetching() -> Tuple[
+    List[NodeInfo],
+    FederationInfo,
+]:
+    """
+    Returns a list of all the nodes in the federation and the whole information for the federation(data_models, datasets).
```

Comment:
`Returns a list of all the nodes in the federation and their metadata (data_models, datasets, cdes).` ??